File: YeChanSong_코딩테스트_연습/정렬/H-index.py
def solution(init):
    answer = 0
    srtd = sorted(init)
    n = len(srtd)
    h = 0
    for i in range(n):
        while 1:
            if h > srtd[i]:
                break
            if h <= srtd[i] and n - i >= h:    
                answer = h
            h +=1
    return answer

File: YeChanSong_코딩테스트_연습/정렬/H-index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer = 0, cnt=0, n = citations.length;
        Arrays.sort(citations);
        for(int i=0;i<citations[n-1];i++){
            cnt=0;
            for(int j=0; j<n; j++){
                if(citations[j]>=i)
                    cnt++;
                if(cnt==i)
                    break;
            }
            if(cnt<i)
                break;
            answer =i;
        }
        return answer;
    }
}